Paco-Server/ear/default/web/js/viz/vizService.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Created by muthuk on 6/28/17.
+ */
+pacoApp.factory('experimentsVizService', ['$http', 'experimentService', function ($http, experimentService) {
+
+    var expObject = '';
+
+    function getExperimentObj(id) {
+
+        var getExp = experimentService.getExperiment(id).then(function (experimentValue) {
+            return experimentValue.data;
+        }, function (error) {
+            console.log(error);
+        });
+
+        expObject = getExp.then(function (experimentObj) {
+            return experimentObj.results[0];
+        }, function (error) {
+            console.log(error);
+        });
+        return expObject;
+    }
+
+    function getEventsResponseDetails(experimentObject) {
+
+        var responseDetails = [];
+        var inputs = experimentObject.groups[0].inputs;
+        inputs.forEach(function (input) {
+            if (input.responseType == "likert") {
+                responseDetails.push({
+                    "name": input.name,
+                    "responseType": input.responseType,
+                    "text": input.text,
+                    "leftsidelabel": input.leftSideLabel,
+                    "rightsidelabel": input.rightSideLabel
+                });
+            } else if (input.responseType == "list") {
+                responseDetails.push({
+                    "name": input.name,
+                    "responseType": input.responseType,
+                    "text": input.text,
+                    "listChoices": input.listChoices
+                });
+            } else {
+                responseDetails.push({"name": input.name, "responseType": input.responseType, "text": input.text});
+            }
+        });
+        return responseDetails;
+    }
+
+    function getExperimentEvents(id) {
```

Comment:
id -> experimentId
Also, I think it is more concise to name it getEvents since only experiments have events :-)